src/handle-data.ts

```diff
@@ -13,29 +17,31 @@ const MAX_CELLS = 10000;
  * @param {Number} numDims
  * @returns dimensions and measures total cell values as strings
  */
-export function getTotalInfo(isDim, layout, colIndex, numDims) {
+export function getTotalInfo(isDim: boolean, layout: TableLayout, colIndex: number, numDims: number) {
   if (!isDim) return layout.qHyperCube.qGrandTotalRow[colIndex - numDims]?.qText;
   if (colIndex !== 0) return '';
   return layout.totals.label;
 }
 
-export function getHighestPossibleRpp(width, rowsPerPageOptions) {
+export function getHighestPossibleRpp(width: number, rowsPerPageOptions: number[]) {
   const highestPossibleOption = [...rowsPerPageOptions].reverse().find((opt) => opt * width <= MAX_CELLS);
   return highestPossibleOption || Math.floor(MAX_CELLS / width); // covering corner case of lowest option being too high
 }
 
-export function getColumnOrder({ qColumnOrder, qDimensionInfo, qMeasureInfo }) {
+export function getColumnOrder({ qColumnOrder, qDimensionInfo, qMeasureInfo }: HyperCube) {
   const columnsLength = qDimensionInfo.length + qMeasureInfo.length;
   return qColumnOrder?.length === columnsLength ? qColumnOrder : [...Array(columnsLength).keys()];
 }
 
-export function getColumnInfo(layout, colIndex) {
+export function getColumnInfo(layout: TableLayout, colIndex: number) {
   const { qDimensionInfo, qMeasureInfo } = layout.qHyperCube;
   const numDims = qDimensionInfo.length;
   const isDim = colIndex < numDims;
-  const info = isDim ? qDimensionInfo[colIndex] : qMeasureInfo[colIndex - numDims];
+  const info = (isDim ? qDimensionInfo[colIndex] : qMeasureInfo[colIndex - numDims]) as
+    | ExtendedNxMeasureInfo
+    | ExtendedNxDimensionInfo;
   const isHidden = info.qError?.qErrorCode === 7005;
-  const isLocked = info.qLocked;
+  const isLocked = isDim && qDimensionInfo[colIndex].qLocked;
```

Comment:
why can't you use info here

Reply:
as qLocked does not exist in qMeasureInfo

Reply:
but it will never be qMeasureInfo if isDim is true. Does
```suggestion
  const isLocked = isDim && info.qLocked;
```
work or does typescript complain?

Reply:
yes, typescript does not know that, so it complains

Reply:
```suggestion
  const isLocked = isDim && (info as ExtendedNxDimensionInfo).qLocked;
```